### PR TITLE
NACK parsing

### PR DESCRIPTION
**Pull request by bblaszkow06**

### CODE REVIEW

**Review by mickel8 — APPROVED** [review]

File: lib/membrane/rtcp/transport_feedback_packet/nack.ex [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      |> Enum.reject(&(&1 === nil))
+    end
+    |> then(&{:ok, %__MODULE__{lost_packet_ids: List.flatten(&1)}})
+  end
+
+  @impl true
+  def encode(%__MODULE__{lost_packet_ids: lost_packet_ids}) do
+    ids_to_encode = Enum.sort(lost_packet_ids)
+
+    # code splitting ids into tuples with reference_id and a list of following ids
```
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply lib/membrane/rtcp/transport_feedback_packet/nack.ex:73]: With that solution you may put additional packets as missing - there's no check if reference_id is part of lost sequence numbers, i.e. if `lost = [1, 20]` that code would generate 2 FCIs: `<<1::16, 0::16>>` and `<<17::16, 0b100::16>>` reporting additional missing id - 17.
This however led me to another problem existing in both approaches - seq_num wrapping. The example for that is `<<65535::16, 1::16>> => [65535, 0]`

File: lib/membrane/rtcp/transport_feedback_packet/nack.ex [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  def decode(nack_fci) do
+    for <<packet_id::unsigned-size(16), bit_mask::16-bits <- nack_fci>> do
+      next_lost_packets =
+        for(<<bit::1 <- bit_mask>>, do: bit)
+        |> Enum.reverse()
+        |> Enum.with_index(1)
+        |> Enum.map(fn
+          {1, position} -> position + packet_id
+          {0, _position} -> nil
+        end)
```
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply lib/membrane/rtcp/transport_feedback_packet/nack.ex]: I created a bit of a hybrid - I wanted to limit the integer/bitwise magic to the minimum and keep the explicit `nil`

File: lib/membrane/rtcp/transport_feedback_packet/nack.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (mickel8) [reply lib/membrane/rtcp/transport_feedback_packet/nack.ex:76]: I see, I think we should leave FIXME tag with explanation here

**Review by mickel8 — APPROVED** [review]

File: lib/membrane/rtcp/transport_feedback_packet/nack.ex [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  def decode(nack_fci) do
+    for <<packet_id::unsigned-size(16), bit_mask::unsigned-size(16) <- nack_fci>> do
+      next_lost_packets =
+        0..15
+        |> Enum.map(fn i ->
+          if (bit_mask >>> i &&& 1) == 1 do
+            mod_16bit(packet_id + i + 1)
+          else
+            nil
+          end
```
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply lib/membrane/rtcp/transport_feedback_packet/nack.ex:63]: IIUC you mean:
```suggestion
          if (bit_mask >>> i &&& 1) == 1 do
            mod_16bit(packet_id + i + 1)
          end
```

**Pull request merged**